scio-elasticsearch/src/main/scala/com/spotify/scio/elasticsearch/package.scala

```diff
@@ -82,13 +74,13 @@ package object elasticsearch {
      * @param f function to transform arbitrary type T to Elasticsearch [[ActionRequest]]
      * @param numOfShard number of parallel writes to be performed, recommended setting is the
      *                   number of pipeline workers
-     * @param errorHandler function to handle error when performing Elasticsearch bulk writes
+     * @param e function to handle error when performing Elasticsearch bulk writes
      */
     def saveAsElasticsearch(esOptions: ElasticsearchOptions,
-                            f: T => Iterable[ActionRequest[_]],
                             flushInterval: Duration = Duration.standardSeconds(1),
-                            numOfShard: Long,
-                            errorHandler: BulkExecutionException => Unit): Future[Tap[T]] = {
+                            numOfShard: Long = defaultNumOfShards())
+                           (f: T => Iterable[ActionRequest[_]],
+                            e: BulkExecutionException => Unit = m => throw m): Future[Tap[T]] = {
```

Comment:
Curry `f` and `e` so that one can write `.saveAsElasticsearch(options) { t => ... }`?
Or maybe put `e` in the first group so it's `(esOptions, flushInterval, numOfShards, e)`?
While you're at it, rename `numOfShard` => `numOfShards` so it's consistent with other IOs.